bears/coffee_script/CoffeeLintBear.py

```diff
@@ -22,11 +23,280 @@ class CoffeeLintBear:
+                        braces_spacing_width: int=1,
+                        spacing_in_empty_braces: int=1,
+                        class_naming_camel_case: bool=True,
+                        spaces_before_colon: int=-1,
+                        spaces_after_colon: int=-1,
+                        enforce_newline_at_EOF: bool=True,
+                        use_spaces: bool=True,
+                        tab_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
+                        number_of_newlines_after_classes: int=-1,
+                        no_backticks: bool=True,
+                        no_implicit_braces: bool=False,
+                        no_implicit_parentheses: bool=False,
+                        no_interpolation_in_single_quotes: bool=True,
+                        no_stand_alone_at: bool=True,
+                        disable_throwing_strings: bool=True,
+                        allow_trailing_semicolons: bool=False,
+                        allow_trailing_whitespaces: bool=False,
+                        no_unnecessary_double_quotes: bool=False,
+                        use_english_operator: bool=False,
+                        spaces_around_operators: bool=False,
+                        space_after_comma: bool=True,
+                        cyclomatic_complexity: int=10,
+                        prevent_duplicate_keys: bool=True,
+                        consistent_line_endings_style: str='',
+                        no_this: bool=False,
+                        no_decr_or_incrementation_operators: bool=False,
+                        no_empty_parameter_list: bool=False,
+                        no_empty_functions: bool=False,
+                        enforce_parentheses_on_non_empty_constructors: bool=True
+                        ):
+        """
+        :param max_line_length:
+            Maximum number of characters per line.
+        :param max_line_length_affect_comments:
+            Determines if ``max_line_length`` should also affects comments or
+            not.
+        :param space_before_and_after_arrow:
+            Determines if spaces should be used before and after the arrow.
+        :param check_braces_spacing:
+            Checks if proper spacing is used inside curly braces.
+        :param braces_spacing_width:
+            Determines the number of blank spaces after the opening ``{`` and
+            before the closing brace ``}`` given that there is something within
+            the braces.
+        :param spacing_in_empty_braces:
+            Determines the number of blank spaces after the opening ``{`` and
+            before the closing brace ``}`` given empty content.
+        :param class_naming_camel_case:
+            Checks whether the classes name should be in camel-case or not.
+        :param spaces_before_colon:
+            Determines the number of blank spaces before colon.
+        :param spaces_after_colon:
+            Determines the number of space after colon.
```

Comment:
essentially -1 means "some default value but I don't show it to you :P"
which is hiding information from the user, right? Why not put the default
there explicitly?

Sincerely,

Lasse Schuirmann

lasse@schuirmann.net
http://coala-analyzer.org/ - http://viperdev.io/ - http://gitmate.com/

On 14 July 2016 at 19:11, Asnelchristian <notifications@github.com> wrote:

> In bears/coffee_script/CoffeeLintBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/555#discussion_r70844403>
> :
>
> > +            Determines if spaces should be used before and after the arrow.
> > +        :param check_braces_spacing:
> > +            Checks if proper spacing is used inside curly braces.
> > +        :param braces_spacing_width:
> > +            Determines the number of blank spaces after the opening ``{`` and
> > +            before the closing brace ``}`` given that there is something within
> > +            the braces.
> > +        :param spacing_in_empty_braces:
> > +            Determines the number of blank spaces after the opening ``{`` and
> > +            before the closing brace ``}`` given empty content.
> > +        :param class_naming_camel_case:
> > +            Checks whether the classes name should be in camel-case or not.
> > +        :param spaces_before_colon:
> > +            Determines the number of blank spaces before colon.
> > +        :param spaces_after_colon:
> > +            Determines the number of space after colon.
>
> the default value for this config (the one provided by the linter ) is
> used unless the user modify one the those. -1 simply because they can take
> any natural number
>
> Hope it clear it out. or do you think i should remove it? :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/555/files/f0b436c93032c451453f0a650f4c8c342e427fa3#r70844403>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFc6KEyhzMMlOqjm8XVy_buKg3KqP5Dbks5qVm3SgaJpZM4I9f6t>
> .
>